server/interpreter/data/object.go

```diff
@@ -16,11 +16,48 @@
 
 package data
 
-// Object represents typical JavaScript objects with (optional)
-// prototype, properties, etc.
-type Object struct {
+// Object represents any JavaScript value (primitive, object, etc.).
+type Object interface {
+	/// Any Object is a valid Value.
+	Value
+
+	// Proto returns the prototype (parent) object for this object.
+	// N.B. this is object.__proto__, not Constructor.prototype!
+	Proto() Object
+
+	// Get returns the current value of the given property or an
+	// ErrorMsg if that was not possible.
+	Get(name string) (Value, *ErrorMsg)
```

Comment:
Returning an error (rather than a custom error type, like *ErrorMsg) is more idiomatic Go. I realize structured errors can be useful, but usually this is done by casting to a type from error rather than returning that type directly. Also, the ErrorMsg type doesn't look like it has particularly useful fields (do you often extract these fields individually?).
https://github.com/NeilFraser/CodeCity/blob/master/server/interpreter/data/error.go